src/dispatch/incident/views.py

```diff
@@ -138,6 +138,16 @@ def update_incident(
     if not incident:
         raise HTTPException(status_code=404, detail="The requested incident does not exist.")
 
+    # we want to provide additional protections around restricted incidents
+    if incident.visibility == Visibility.restricted:
+        # reject if the user isn't an admin or commander
+        if current_user.email != incident.commander.email or current_user.role != UserRoles.admin:
+            return incident
```

Comment:
We should throw the exception here instead of returning.

Reply:
Fixed.